src/module.rs

```diff
@@ -167,28 +198,28 @@ impl<'a> Module<'a> {
 
 impl<'a> fmt::Display for Module<'a> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let ansi_strings = self.ansi_strings();
-        write!(f, "{}", ANSIStrings(&ansi_strings))
+        for ansi_string in self.ansi_strings() {
+            f.write_str(&ansi_string)?;
+        }
+
+        Ok(())
     }
 }
 
-fn ansi_strings_modified(ansi_strings: Vec<ANSIString>, shell: Shell) -> Vec<ANSIString> {
+fn ansi_strings_modified(ansi_strings: Vec<String>, shell: Shell) -> Vec<String> {
     ansi_strings
         .into_iter()
-        .map(|ansi| {
-            let wrapped = wrap_colorseq_for_shell(ansi.to_string(), shell);
-            ANSIString::from(wrapped)
-        })
-        .collect::<Vec<ANSIString>>()
+        .map(|ansi| wrap_colorseq_for_shell(ansi, shell))
+        .collect::<Vec<String>>()
 }
 
-fn ansi_line<'a, I>(segments: &mut I, term_width: Option<usize>) -> Vec<ANSIString<'a>>
+fn ansi_line<'a, I>(segments: &mut I, term_width: Option<usize>) -> Vec<String>
```

Comment:
Why didn't you use `owo_colors::Styled`, which seems to be the `owo_colors`-equivalent to `AnsiString`? This might also allow keeping the Strings as `Cow<str, 'a>`. Coupling the string and style might allow getting this to work without `strip-ansi-escapes`.